src/objects/repo.ts

```diff
@@ -51,12 +51,13 @@ export class ObjectRepository<T extends PayloadModel> {
    * @param data data to be saved
    */
   async create(owner: string, data: Partial<T>): Promise<T> {
-    // @ts-ignore bad typescript
-    return this.internalRepo.create({
-      object_state: ObjectState.Created,
-      __owner: owner,
-      ...data
-    });
+    return (
+      await this.internalRepo.create({
+        object_state: ObjectState.Created,
+        __owner: owner,
+        ...data
+      }))
+      .toObject();
```

Comment:
this works, but please simplify your code. there's no need for everything to be on one line